pkg/clients/aws.go

```diff
@@ -169,8 +183,52 @@ func Int64Value(v *int64) int {
 	return int(aws.Int64Value(v))
 }
 
+// Float64Value converts the supplied float64 pointer to an int, returning zero if
+// the pointer is nil.
+func Float64Value(v *float64) int {
+	return int(aws.Float64Value(v))
+}
+
 // BoolValue converts the supplied bool pointer to a bool, returning false if
 // the pointer is nil.
 func BoolValue(v *bool) bool {
 	return aws.BoolValue(v)
 }
+
+// IntValue converts the supplied int pointer to a int, returning 0 if
+// the pointer is nil.
+func IntValue(v *int) int {
+	return aws.IntValue(v)
+}
+
+func LateInitializeStringPtr(in *string, from *string) *string {
+	if in != nil {
+		return in
+	}
+	return from
+}
+
+func LateInitializeBoolPtr(in *bool, from *bool) *bool {
+	if in != nil {
+		return in
+	}
+	return from
+}
+
+func LateInitializeBool(in *bool, from bool) *bool {
+	if in != nil {
+		return in
+	}
+	return &from
+}
```

Comment:
https://github.com/crossplaneio/stack-gcp/blob/master/pkg/clients/gcp.go#L247

This looks like it differs logically from the GCP implementation. I believe the intent of all the late init functions are:

1. Return `in` if `in` is not `nil` or if `from` is its zero value.
1. Return `from`.

This implementation would return a pointer to a `false` bool, which I believe will be interpreted as the optional value being set to `false`, not unset.

Reply:
I think ignoring the zero-value may not make sense in all cases. I wonder if we should make them `func(pointer, pointer) pointer` or `func(pointer, pointer) value`. These 2 functions for each basic type would cover the cases where we call late-initialization functions:
```
pointer <-> pointer
pointer <-> value
value <-> pointer
value <-> value
```
The `value` inputs can easily be given as `&value` while the inverse wouldn't make as much sense. If the value really is `false`, we'd take that into account and set the value to false.

BTW, the reason I needed `LateInitializeBool` was the function `automaticFailoverEnabled` that returns `bool`. But taking a second look, it should actually return `*bool` because failover status string could be empty. So, we don't need `LateInitializeBool` anymore.